### ISSUE TITLE
RuleEngineCallbackImpl throws NPE during unit tests

### ISSUE THREAD

**Issue by kaikreuzer** [eclipse/smarthome#1443]
A unit test fails because of an NPE, see https://hudson.eclipse.org/smarthome/job/SmartHomeDistribution-Nightly/1003/testReport/junit/org.eclipse.smarthome.automation.module.timer.internal/RuntimeRuleTest/assert_that_timerTrigger_works/:

```
16:09:49.016 [DefaultQuartzScheduler_Worker-2] ERROR org.quartz.core.JobRunShell - Job DEFAULT.TimerTrigger52848043-4d61-40f7-96ca-ed4eac51aa00 threw an unhandled Exception: 
java.lang.NullPointerException: null
	at org.eclipse.smarthome.automation.core.internal.RuleEngineCallbackImpl.triggered(RuleEngineCallbackImpl.java:44) ~[na:na]
	at org.eclipse.smarthome.automation.module.timer.handler.CallbackJob.execute(CallbackJob.java:53) ~[na:na]
	at org.quartz.core.JobRunShell.run(JobRunShell.java:202) ~[quartz-2.2.1.jar:na]
	at org.quartz.simpl.SimpleThreadPool$WorkerThread.run(SimpleThreadPool.java:573) [quartz-2.2.1.jar:na]
16:09:49.019 [DefaultQuartzScheduler_Worker-2] ERROR org.quartz.core.ErrorLogger - Job (DEFAULT.TimerTrigger52848043-4d61-40f7-96ca-ed4eac51aa00 threw an exception.
org.quartz.SchedulerException: Job threw an unhandled exception.
	at org.quartz.core.JobRunShell.run(JobRunShell.java:213) ~[quartz-2.2.1.jar:na]
	at org.quartz.simpl.SimpleThreadPool$WorkerThread.run(SimpleThreadPool.java:573) [quartz-2.2.1.jar:na]
Caused by: java.lang.NullPointerException: null
	at org.eclipse.smarthome.automation.core.internal.RuleEngineCallbackImpl.triggered(RuleEngineCallbackImpl.java:44) ~[na:na]
	at org.eclipse.smarthome.automation.module.timer.handler.CallbackJob.execute(CallbackJob.java:53) ~[na:na]
	at org.quartz.core.JobRunShell.run(JobRunShell.java:202) ~[quartz-2.2.1.jar:na]
	... 1 common frames omitted
```

**Comment by kaikreuzer** [comment]
I have to correct myself: The NPE is not the cause of the failing test.
I have created https://github.com/eclipse/smarthome/pull/1444 to fix avoid the NPE, but the test still fails. As it seems to be failing since https://github.com/eclipse/smarthome/pull/1434, I have just reverted #1434 again to make the build green again until a fix is found. 
@adimova Did the tests succeed locally for you? Could you investigate why this change makes the test fails?

**Comment by adimova** [comment]
Hi  @kaikreuzer 
There are two reasons for test to fail:
1) There is a bug in TimerModuleHandlerFactory.internalCreate(Module module, String ruleUID)
2) ManagedRuleProvider is registered as a service and RuleRegistry is added as a ProviderChangeListener of ManagedRuleProvider

The first will be fixed as well. The question is do we want to keep the ability to add new rules, using ManagedRuleProvider.add(E element)?

**Comment by kaikreuzer** [comment]
If possible, yes - at least we are using the managed providers directly in many unit tests. It should just work the same way as any other provider that might during runtime change the set of rules it provides (this is what #905 is all about).

**Comment by danchom** [comment]
implemented by #1500
